ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/admin/ApiPartyManagementService.scala

```diff
@@ -47,14 +47,50 @@ class ApiPartyManagementService private (
       .listParties()
       .map(ps => ListKnownPartiesResponse(ps.map(mapPartyDetails)))(DE)
 
+  /**
+    * Continuously polls the party management service to check if a party has been persisted.
+    *
+    * Despite the `go` inner function not being stack-safe per se, only one stack frame will be on
+    * the stack at any given time since the "recursive" invocation happens on a different thread.
+    *
+    * @param party The party whose persistence we're waiting for
+    * @return The number of attempts before the party was found wrapped in a [[Future]]
+    */
+  private def pollUntilPersisted(party: String): Future[Int] = {
```

Comment:
Should this function have exponential backoff and a maximum wait time?

Reply:
Regarding maximum wait time, I'm not entirely sure. The polling here is done to ensure that the data is available to be read, but this depends on the indexer rather then the underlying service, so failing to read back from the indexer storage would result in a "failed write" even though the data at this point should definitely have been persisted on the participant. I would agree on this point but we'd have to make sure that we make party allocation idempotent. Would this be possible?

On the back-off strategy I agree, but what would you think would be reasonable values? Should we somehow make this configurable?

@gerolf-da we could probably benefit if you could share your thoughts.

Reply:
Exponential back-off addressed by https://github.com/digital-asset/daml/pull/2040/commits/b185bea83b76e7e3f6528b98c0d6941d0f3be4ac

Reply:
I would leave maximum waiting time out for now but it's definitely worth discussing this, otherwise this is still error prone. We must have a clear idea of the write semantics. If we can pursue the idempotent strategy it would probably be the simplest course of action.